src/main/java/net/dries007/tfc/objects/te/TESluice.java

```diff
@@ -104,17 +104,24 @@ public void update()
                     }
                     if (Constants.RNG.nextDouble() < ConfigTFC.Devices.SLUICE.gemChance)
                     {
-                        Gem dropGem;
-                        if (Constants.RNG.nextDouble() < ConfigTFC.Devices.SLUICE.diamondGemChance)
+                        Chunk chunk = world.getChunk(pos);
```

Comment:
My only concern here is that this will only check and work on the sluice's chunk, check the above lines (L73-L87) for how to do it. I also recommend moving those lines to a new function so it can be used by both ores and gems, just don't forget to add a boolean parameter for ores, since those chunks only are worked if there is a vein there.